### PR TITLE
Adding tests and fixing bugs in importance mask mode

### PR DESCRIPTION
**Pull request by dzenanz**

### CODE REVIEW

**Comment by dzenanz** [comment]
I had already started working on adding automated regression tests for importance mask mode. Fixes for bugs introduced by @lassoan are interspersed.

File: src/PlusVolumeReconstruction/Testing/CMakeLists.txt [hunk 1]
```diff
@@ -1,104 +1,33 @@
...
-    ${PLUS_EXECUTABLE_OUTPUT_PATH}/VolumeReconstructor
-    --config-file=${ConfigFilesDir}/Testing/PlusDeviceSet_VolumeReconstructionOnly_SonixRP_TRUS_D70mm_LN_MEAN.xml
-    --source-seq-file=${TestDataDir}/SpinePhantomFreehand.mha
-    --output-volume-file=vtkVolumeReconstructorTestLNMEANvolume.mha
-    --image-to-reference-transform=ImageToReference
-    )
-  SET_TESTS_PROPERTIES( vtkVolumeReconstructorTestRunLinrMeanUChar PROPERTIES FAIL_REGULAR_EXPRESSION "ERROR;WARNING" )
-
-  ADD_TEST(vtkVolumeReconstructorTestRunLinrMaxiUChar
+function(RegressionTest TestName Config Seq Out)
```
Comment (adamrankin) [inline src/PlusVolumeReconstruction/Testing/CMakeLists.txt]: Niiiiccceeee. +1

**Comment by adamrankin** [comment]
@lassoan Leaving it to you to merge.

File: src/PlusVolumeReconstruction/Testing/CMakeLists.txt (hunk 1 above)
Comment (dzenanz) [inline src/PlusVolumeReconstruction/Testing/CMakeLists.txt]: That was not just hard to look at, but also not very conductive of uniformity :smile:

Comment (dzenanz) [inline src/PlusVolumeReconstruction/Testing/CMakeLists.txt]: Moreover, this change keeps the old test names.

File: src/PlusVolumeReconstruction/vtkPlusPasteSliceIntoVolumeHelperCommon.h [hunk 2]
```diff
@@ -344,6 +344,7 @@ static int vtkTrilinearInterpolation(F *point,
           break;
         }
         inPtrTmp++;
+        importancePtr++;
```
Comment (lassoan) [inline src/PlusVolumeReconstruction/vtkPlusPasteSliceIntoVolumeHelperCommon.h]: Are you sure you want to increment importancePtr for each scalar component?
As I saw, importance image is always a single-component image, while input and output may have multiple components.

Comment (dzenanz) [inline src/PlusVolumeReconstruction/vtkPlusPasteSliceIntoVolumeHelperCommon.h]: You are right. It does not need to be modified at all, as this whole function processes just one input pixel (and each importance mask pixel corresponds to one input pixel).

**Comment by dzenanz** [comment]
Don't merge this just yet. I discovered more bugs while adding tests.

**Comment by dzenanz** [comment]
I also added required data and configuration files to PlusLibData. Time to review and merge.

File: src/PlusVolumeReconstruction/vtkPlusPasteSliceIntoVolumeHelperUnoptimized.h [hunk 3]
```diff
@@ -358,7 +358,7 @@ static void vtkUnoptimizedInsertSlice(vtkPlusPasteSliceIntoVolumeInsertSlicePara
   {
     for (int idY = inExt[2]; idY <= inExt[3]; idY++, inPtr += inIncY, importancePtr += impIncY)
     {
-      for (int idX = inExt[0]; idX <= inExt[1]; idX++, inPtr += numscalars, importancePtr += impIncX)
+      for (int idX = inExt[0]; idX <= inExt[1]; idX++, inPtr += numscalars, importancePtr += 1)
```
Comment (lassoan) [inline src/PlusVolumeReconstruction/vtkPlusPasteSliceIntoVolumeHelperUnoptimized.h]: I'm not sure if it makes any difference in performance, but it's a hot loop so maybe it's better to write importancePtr++

Comment (dzenanz) [inline src/PlusVolumeReconstruction/vtkPlusPasteSliceIntoVolumeHelperUnoptimized.h]: Any half-baked optimizing compiler will do that by itself :smile:

**Pull request merged**